test/DataProvider/MessageDataProvider.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace PhoneBurnerTest\Http\Message\DataProvider;
+
+use Psr\Http\Message\StreamInterface;
+
+trait MessageDataProvider
+{
+    public function provideAllMethods(): \Generator
+    {
+        foreach ($this->provideWithMethods() as $label => $data) {
+            yield $label => $data;
+        }
+
+        foreach ($this->provideGetterMethods() as $label => $data) {
+            yield $label => $data;
+        }
+    }
+
+    public function provideGetterMethods(): \Generator
+    {
+        yield "getProtocolVersion() => '1.1'" => ['getProtocolVersion', [], '1.1'];
+        yield "getProtocolVersion() => '1.0'" => ['getProtocolVersion', [], '1.0'];
+
+        $header = [
+            'line one',
+            'line two'
```

Comment:
MUST: add trailing comma in multi-line array
```suggestion
            'line two',
```